src/database/GameIds.ts

```diff
@@ -87,4 +105,42 @@ export class GameIds extends EventEmitter {
     }
     return {games: this.games, participantIds: this.participantIds};
   }
+
+  public mark(gameId: GameId) {
+    console.log(`Marking ${gameId} to be evited in ${this.config.evictMillis}ms`);
+    this.evictionSchedule.set(gameId, this.clock.now() + this.config.evictMillis);
+  }
+
+  public sweep() {
+    console.log('Starting sweep');
+    const now = this.clock.now();
+    for (const entry of this.evictionSchedule.entries()) {
+      if (entry[1] <= now) {
+        const gameId = entry[0];
+        console.log(`evicting ${gameId}`);
+        this.evict(gameId);
+        this.evictionSchedule.delete(gameId);
+      }
+    }
+    console.log('Finished sweep');
+  }
+
+  private evict(gameId: GameId) {
+    const game = this.games.get(gameId);
+    if (game === undefined) return;
+    game.getPlayers().forEach((p) => (p as any)._game = undefined);
```

Comment:
```suggestion
    game.getPlayers().forEach((p) => p.game = undefined);
```

This will involve updating the setter for `game` to take `Game | undefined` as well. Will avoid using `any` and we allow `game` to be `undefined` on `Player`.

Reply:
Yeah, sorry. I fixed it. PTAL. Making game take undefined will be a huge maintenance issue. A proper fix is to define Game before Player, and not the other way around.

There's another possibility which relies on `GameLoader.getGame` but that's slower and messier. A proper refactor seems the right step forward.

Reply:
The solution I will push shortly seems the right compromise.